homeassistant/components/androidtv/config_flow.py

```diff
@@ -0,0 +1,300 @@
+            )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors={"base": error},
+        )
+
+    async def _async_check_connection(self, user_input):
+        """Attempt to connect the Android TV."""
+
+        try:
+            aftv, message = await async_connect_androidtv(self.hass, user_input)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with Android TV at %s", user_input[CONF_HOST]
+            )
+            return RESULT_UNKNOWN, None
+
+        if not aftv:
+            _LOGGER.warning(message)
+            return RESULT_CONN_ERROR, None
+
+        dev_prop = aftv.device_properties
+        unique_id = format_mac(
+            dev_prop.get(PROP_ETHMAC) or dev_prop.get(PROP_WIFIMAC, "")
+        )
+        await aftv.adb_close()
+        return None, unique_id
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        error = None
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            adb_key = user_input.get(CONF_ADBKEY)
+            adb_server = user_input.get(CONF_ADB_SERVER_IP)
+
+            if adb_key and adb_server:
+                return self._show_setup_form(user_input, "key_and_server")
+
+            if adb_key:
+                isfile = await self.hass.async_add_executor_job(_is_file, adb_key)
+                if not isfile:
+                    return self._show_setup_form(user_input, "adbkey_not_file")
+
+            ip_address = await self.hass.async_add_executor_job(_get_ip, host)
+            if not ip_address:
+                return self._show_setup_form(user_input, "invalid_host")
+
+            self._async_abort_entries_match({CONF_HOST: host})
+            if ip_address != host:
+                self._async_abort_entries_match({CONF_HOST: ip_address})
+
+            error, unique_id = await self._async_check_connection(user_input)
+            if error is None:
+                if not unique_id:
+                    return self.async_abort(reason="invalid_unique_id")
+
+                await self.async_set_unique_id(unique_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input.get(CONF_NAME) or host,
+                    data=user_input,
+                    options=self._import_options,
+                )
+
+        user_input = user_input or {}
+        return self._show_setup_form(user_input, error)
+
+    async def async_step_import(self, import_config=None):
+        """Import a config entry."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_HOST] == import_config[CONF_HOST]:
+                _LOGGER.warning(
+                    "Already configured. This yaml configuration has already been imported. Please remove it"
```

Comment:
Would be nice to tell them for which host, but it doesn't really matter since its going to go away anyways